### DISCUSSION TITLE
Deeper integration with mesa

### DISCUSSION THREAD

**Discussion by adamamer20** [projectmesa/mesa-frames#123] [Ideas]
Currently there are some features that are only available in mesa (eg. visualization). While we work on refactoring those, we could have a "fallback" to convert on-the-fly a mesa-frames object to a mesa object and perform the operation. Performance would take a significant hit but it would allow for an easier transition from mesa.

**Comment by rht** [comment]
The data to be fed to the Solara visualization is simply a list of list, or a 2D NumPy array. The [relevant code](https://github.com/projectmesa/mesa/blob/1202e2e28a3bfc00f285b66087bd2ab8efce1cc1/mesa/visualization/components/matplotlib.py#L26-L60) is only <50 LOC. So it should take about a day to implement. It is a Matplotlib scatter plot.

**Comment by adamamer20** [comment]
> The data to be fed to the Solara visualization is simply a list of list, or a 2D NumPy array. The [relevant code](https://github.com/projectmesa/mesa/blob/1202e2e28a3bfc00f285b66087bd2ab8efce1cc1/mesa/visualization/components/matplotlib.py#L26-L60) is only <50 LOC. So it should take about a day to implement. It is a Matplotlib scatter plot.

I see, seems simple enough. I should look up a bit how solara works though.

**Comment by rht** [comment]
You can assign this to me. Make sure we get Sugarscape for GSoC.
